### ISSUE TITLE
Is it possible to share the mat file for training too? 

### ISSUE THREAD

**Issue by Skorkmaz88** [harryjdavies/HeartGPT#1]
Is it possible to share the mat file for training too, the one used in pretraining example ?

**Comment by ZhijiangTang** [comment]
I am also curious about the processing of the training data. The original article mentioned "Once tokenised, this dataset resulted in over 42 million tokens for training." After reading the code, the author embeds the ECG into a vocabulary of 101 tokens, but how are these 101 tokens classified? Because as far as I know, the ECG is patched and these patches should be different. If pre-training is used, should MSE be used instead of cross entropy loss function?
![image](https://github.com/user-attachments/assets/226e4b0a-0990-4cd7-a36f-66b372358a64)
![image](https://github.com/user-attachments/assets/c5797053-6c80-484a-a996-be07d4a62574)

**Comment by harryjdavies** [comment]
Hi @Skorkmaz88 and @RiverOtherSide , I have uploaded the exact matlab code I used to form the ECG training dataset (and tokenise it), with the link to download the data. The script is called "tokenise_ecg_training_data.m". I don't think I can share the specific dataset files as they are not my datasets.
@RiverOtherSide The ECG is not patched, each time point is an individual token. It is trained to predict the next token with cross entropy loss and 101 classes (similar to NLP). In the ECG dataset, there were 42,000 subjects with 10 seconds of ECG data each. At 100Hz that is 42000x10x100 = 42 million time points / tokens.
As a side note, it is our experience that this is why the attention mechanism is so interpretable in our network - interpretability does not arise if we train with MSE, and it becomes even less interpretable if we use patches.
Happy to clarify further if necessary

**Comment by ZhijiangTang** [comment]
Thank you very much for your answer @harryjdavies . If each point is used as a token and divided into integers ranging from 1 to 100, a vocabulary of size 101 can indeed be formed. Thank you very much for your answer again. Now I have a better understanding of your method of processing data.😊

**Comment by harryjdavies** [comment]
No problem @RiverOtherSide , thank you for your interest in our work! I hope it is useful! I will close this issue now, but feel free to reach out again if any more questions arise
